File: LeetCode/CPP/Easy/1323. Maximum 69 Number.cpp
// Problem Link:
// https://leetcode.com/problems/maximum-69-number/

// Solution:
class Solution {
public:
    int maximum69Number (int num) {
        int flag = 1;
        string s, temp = to_string(num);
        for(int i = 0; i < temp.length(); i++) {
            if(temp[i] == '6' and flag == 1) {
                flag = 0;
                s.push_back('9');
            }
            else
                s.push_back(temp[i]);
        }
        return stoi(s);
    }
};

File: LeetCode/PYTHON/Easy/1323. Maximum 69 Number.py
# Problem Link:
# https://leetcode.com/problems/maximum-69-number/

# Solution:
class Solution:
    def maximum69Number (self, num: int) -> int:
        temp = 0
        six = -1
        n = num
        while n:
            if n % 10 == 6:
                six = temp
            n //= 10
            temp += 1
        return num if six == -1 else num + 3 * 10 ** six